node/impl/full/eth.go

```diff
@@ -1433,9 +1433,12 @@ func (a *EthModule) applyMessage(ctx context.Context, msg *types.Message, tsk ty
 	}
 
 	if res.MsgRct.ExitCode.IsError() {
-		reason := parseEthRevert(res.MsgRct.Return)
-		return nil, xerrors.Errorf("message execution failed: exit %s, revert reason: %s, vm error: %s", res.MsgRct.ExitCode, reason, res.Error)
+		return nil, ethtypes.NewExecutionRevertedWithDataError(
+			errcodeDefault,
```

Comment:
Shouldn't we pass the exit code on the receipt here ? Why are we passing `errcodeDefault` here ?

Reply:
@aarshkshah1992 actually there was some confusion about what error code to return because, in ethereum there two different kind of error codes for the same message (execution reverted), In `EthCall` and `EstimateGas` it returns error code `3` and generally while executing message it returns `-3200`.

I will the default error code to `3` for  `EthCall` and `EstimateGas` which we were already returning in `ErrorCode()`